### PR TITLE
variable: group same kind of variables into structs.

### PR DESCRIPTION
**Pull request by coocood**
Make it organized and easy to write verify test.

### CODE REVIEW

**Comment by coocood** [comment]
@zz-jason @XuHuaiyu PTAL

File: sessionctx/variable/varsutil_test.go [hunk 1]
```diff
@@ -74,6 +75,17 @@ func (s *testVarsutilSuite) TestNewSessionVars(c *C) {
 	c.Assert(vars.MemQuotaIndexLookupReader, Equals, int64(DefTiDBMemQuotaIndexLookupReader))
 	c.Assert(vars.MemQuotaIndexLookupJoin, Equals, int64(DefTiDBMemQuotaIndexLookupJoin))
 	c.Assert(vars.MemQuotaNestedLoopApply, Equals, int64(DefTiDBMemQuotaNestedLoopApply))
+
+	assertFieldsGreaterThanZero(c, reflect.ValueOf(vars.Concurrency))
+	assertFieldsGreaterThanZero(c, reflect.ValueOf(vars.MemQuota))
```
Comment (zz-jason) [inline sessionctx/variable/varsutil_test.go]: memory quota can be smaller than 0, which means the memory usage is not limited

Comment (coocood) [reply sessionctx/variable/varsutil_test.go]: It only verifies that the return value of `NewSessionVars`.

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -584,3 +543,60 @@ type TableDelta struct {
...
+	HashJoinConcurrency int
+
+	// IndexSerialScanConcurrency is the number of concurrent index serial scan worker.
+	IndexSerialScanConcurrency int
+}
+
+// MemQuota defines memory quota values.
+type MemQuota struct {
+	// MemQuotaQuery defines the memory quota for a query.
+	MemQuotaQuery int64
```
Comment (shenli) [inline sessionctx/variable/session.go]: Can we remove the 'Mem' prefix of the variables?

Comment (coocood) [reply sessionctx/variable/session.go]: We embed `MemQuota` in `SessionVars`, so we access `MemQuotaXXX` directly from `SessionVars`, remove `MemQuota` prefix make the variable unclear.

**Pull request merged**